doc/modules/recommended_practices.rst

```diff
@@ -0,0 +1,30 @@
+=====================
+Recommended Practices
+=====================
+
+
+Feature Selection
+=================
+
+**Wrong Practice: Selecting features using Whole Dataset**
+
+    >>> X = numpy.random.randn(500,10000)
+    >>> y = numpy.random.choice(2,500)
+    >>> X_ = SelectKBest(k=25).fit_transform(X,y)
+    >>> print("Mean Accuracy without Pipeline:{0.2f}".format(cross_val_score
+    ...       (GradientBoostingClassifier(),X_,y,cv=3,n_jobs=8).mean()*100))
+    Mean Accuracy without Pipeline: 64.79
```

Comment:
It would be nice to also report the standard deviation of CV scores:

```python
    >>> cv_scores = cross_val_score(GradientBoostingClassifier(),X_selected,y, cv=5)
    >>> print("Cross-validation accuracy: {0.2f} +/- {0.2f}".format(
    ...     cv_scores.mean(), cv_scores.std()))
```

Some additional remarks:

- please do not high values of `n_jobs` in test or examples to avoid oversubscription and excessive memory usage when running the tests/examples on machines that have few CPU cores,
- `n_jobs=8` is wasteful for `cv=3`: there would be only 3 possible parallel fits in this case,
- the dataset is small so it's fine to use the default value of `n_jobs`,
- let's use `cv=5` to get a more stable std estimate.